packages/frontend/src/features/MetricFlow/hooks/useMetricFlowQueryResultsAPI.ts

```diff
@@ -0,0 +1,99 @@
+import { ApiError } from '@lightdash/common';
+import { useQuery } from 'react-query';
+import { UseQueryOptions } from 'react-query/types/react/types';
+import {
+    createMetricFlowQuery,
+    CreateMetricFlowQueryResponse,
+    getMetricFlowQueryResults,
+    GetMetricFlowQueryResultsResponse,
+    QueryStatus,
+} from '../../../api/MetricFlowAPI';
+
+type ApiRequestsState = Pick<
+    ReturnType<typeof useQuery<GetMetricFlowQueryResultsResponse, ApiError>>,
+    'isLoading' | 'data' | 'error' | 'status'
+> &
+    Pick<
+        ReturnType<typeof useQuery<CreateMetricFlowQueryResponse, ApiError>>,
+        'refetch'
+    >;
+
+export const useMetricFlowQueryResultsAPI = (
+    projectUuid: string | undefined,
+    query?: {
+        metrics: string[];
+        dimensions: string[];
+    },
+    useCreateQueryOptions?: UseQueryOptions<
+        CreateMetricFlowQueryResponse,
+        ApiError
+    >,
+    useResultQueryOptions?: UseQueryOptions<
+        GetMetricFlowQueryResultsResponse,
+        ApiError
+    >,
+): ApiRequestsState => {
+    const metricFlowQuery = useQuery<CreateMetricFlowQueryResponse, ApiError>({
+        queryKey: ['metric_flow_query', projectUuid, query],
```

Comment:
is it ok to serialize an object (`query`) into the query key?

Reply:
Yes, docs -> https://tanstack.com/query/v3/docs/react/guides/query-keys#array-keys